lib/streamlit/runtime/state/query_params.py

```diff
@@ -104,6 +112,9 @@ def _send_query_param_msg(self) -> None:
         ctx = get_script_run_ctx()
         if ctx is None:
             return
+        ctx._final_query_params_used = True
+        ctx.check_both_query_apis_called()
```

Comment:
Shouldn't this be using the method internal to this class?

Reply:
Hm I thought about that but I thought it would be a duplication of this code:
```
ctx = get_script_run_ctx()
        if ctx is None:
            return
```
so I just put that there. I guess I'll use the internal method in this class.